### PR TITLE
Replace `build_e2e.dart` with `devtools_tool serve` command

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Work towards https://github.com/flutter/devtools/issues/6324

### CODE REVIEW

File: tool/lib/commands/serve.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    final remainingArguments = List.of(argResults!.arguments)
+      ..remove(_useLocalFlutterFlag)
+      ..remove(_updatePerfettoFlag)
+      ..remove(_buildAppFlag);
+
+    final localDartSdkLocation = Platform.environment['LOCAL_DART_SDK'];
+    if (localDartSdkLocation == null) {
+      throw Exception('LOCAL_DART_SDK environment variable not set. Please add '
+          'the following to your \'.bash_profile\' or \'.bash_rc\' file:\n'
```
Comment (elliette) [inline tool/lib/commands/serve.dart:84]: nit: `.bashrc`

**Review by elliette — APPROVED** [review]
Approved with small nit

File: tool/lib/commands/serve.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      ..remove(_useLocalFlutterFlag)
+      ..remove(_updatePerfettoFlag)
+      ..remove(_buildAppFlag);
+
+    final localDartSdkLocation = Platform.environment['LOCAL_DART_SDK'];
+    if (localDartSdkLocation == null) {
+      throw Exception('LOCAL_DART_SDK environment variable not set. Please add '
+          'the following to your \'.bash_profile\' or \'.bashrc\' file:\n'
+          'export LOCAL_DART_SDK=<absolute/path/to/my/dart/sdk>');
+    }
```
Comment (kenzieschmoll) [inline tool/lib/commands/serve.dart:86]: @DanTup how can we tweak this exception to be useful for windows? We can do 
```
final  exceptionMessage =  Platform.isWindows ? ... : ...;
throw Exception(exceptionMessage);
```

Comment (DanTup) [reply tool/lib/commands/serve.dart:86]: I would probably just say something like

> Please set the LOCAL_DART_SDK environment variable in Windows to <absolute/path/to/my/dart/sdk>

Unfortunately I've never found a good page on the MS site on how to set environment variables, but there are plenty of unofficial sites with instructions for all versions of Windows.

**Comment by hangyujin** [comment]
CONTRIBUTING.md should also be updated

**Comment by DanTup** [comment]
> CONTRIBUTING.md should also be updated

Thanks! I'll do this soon - I'm going to need to update the VS Code bits (changes to `dart.customDevTools`) anyway, but I might make some additional tweaks after https://github.com/flutter/devtools/pull/6695 lands.

**Pull request merged**